### PR TITLE
Add asset search

### PR DESCRIPTION
**Pull request by Bishoy-at-pieces**
https://drive.google.com/file/d/1zFKEUIq-Q3qhVFAWJA9uviUrK9hLH7ct/view?usp=sharing

### CODE REVIEW

File: search/search_command.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from pieces.assets.list_assets import PiecesAssetIdInputHandler
+
+from pieces_os_client import SearchApi,AssetsApi
+
+
+
+
+class QueryInputHandler(sublime_plugin.TextInputHandler):
+	def placeholder(self):
+		return "Enter a quick description of the bug itself or useful error message"
```
Comment (caleb-at-pieces) [inline search/search_command.py:15]: what is this placeholder for?

File: search/search_command.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		elif search_type == 'fts':
+			api_instance = SearchApi(PiecesSettings.api_client)
+			results = api_instance.full_text_search(query=query)
+			# Check and extract asset IDs from the results
+		if results:
+			# Extract the iterable which contains the search results
+			iterable_list = results.iterable if hasattr(results, 'iterable') else []
+
+			# Check if iterable_list is a list and contains SearchedAsset objects
+			if isinstance(iterable_list, list) and all(hasattr(asset, 'exact') and hasattr(asset, 'identifier') for asset in iterable_list):
```
Comment (caleb-at-pieces) [inline search/search_command.py:75]: good job with null safety!

**Review by caleb-at-pieces — CHANGES_REQUESTED** [review]
it looks like we have an extra placeholder method in one of the classes also we should probably break these classes into their own files

File: search/search_command.py (hunk 1 above)
Comment (Bishoy-at-pieces) [reply search/search_command.py:15]: Yupp mb needs to be changed

**Review by caleb-at-pieces — APPROVED** [review]

File: search/search_command.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+			return self.window.run_command("pieces_list_assets",args={"pieces_asset_id":pieces_asset_id})
+
+	@staticmethod
+	def search(search_type,query)-> list:
+		if search_type == 'assets':
+			api_instance = AssetsApi(PiecesSettings.api_client)
+			results = api_instance.assets_search_assets(query=query, transferables=False)
+		elif search_type == 'ncs':
+			api_instance = SearchApi(PiecesSettings.api_client)
+			results = api_instance.neural_code_search(query=query)
```
Comment (mark-at-pieces) [inline search/search_command.py:65]: might need to specify transferables on this as well but cant remember 100%

**Review by mark-at-pieces — APPROVED** [review]

**Pull request merged**